design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,165 @@
+
+## Background
+
+As a Conjur User or Host, I want to send authentication request to `authn` authenticator (for example).
+So that I receive a JWT access token.
+
+![alt text](encoded_token_chart.jpeg "Login process")
+
+Current authentication response is a Flattened JWS JSON Serialization, e.g:
+```json
+{
+    "protected": "eyJh...",
+    "payload": "eyJz...",
+    "signature": "CZgB..."
+}
+``` 
+
+In order to use this token for restricted endpoint it **must** be encode and added to `Authorization: Token token=` header:
+```sh
+echo "Get Conjur access token using an Azure access token"
+# Get an authn-azure Conjur access token for host azure-apps/test-app
+authn_azure_response=$(curl -k -X POST \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  --data "jwt=$azure_access_token" \
+  https://"$CONJUR_SERVER_DNS":8443/authn-azure/test/cucumber/host%2Fazure-apps%2Ftest-app/authenticate)
+authn_azure_access_token=$(echo -n "$authn_azure_response" | base64 | tr -d '\r\n')
+
+echo "Retrieve a secret using the Conjur access token"
+# Retrieve a Conjur secret using the authn-azure Conjur access token
+secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
+  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
+
+echo "Retrieved secret ${secret} from Conjur!!!"
+```
+[Code reference](https://github.com/cyberark/conjur/blob/application_identity_validation/ci/authn-azure/run-authn-azure.sh#L16)
+
+
+The requested situation is to support both response of current **Flattened** JWS JSON Serialization with content type `application/json` and encoded JWS **Compact** Serialization with content type `text\plain`. 
+
+## Issue description
+1. HTTP server best practices: Using `application\json` as the content type to encoded output (not a JSON) is bad practice.
+    So encoded output shouldn't use `application/json` unless it will be in JSON format. 
+    The `../autheticate` response is in JSON format so it makes sense to use `application\json` as the response content type.
+        As well as `../logic`,  which returns the api key as `text\plain`.
+
+## Solution - Accept HTTP Headers
+According to [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept):
+ > The `Accept` request HTTP header advertises which content types the client is able to understand. Using content negotiation, the server then selects one of the proposals, uses it and informs the client of its choice with the `Content-Type` response header.
+
+To conclude, we can choose the response's `Content-Type` by the request's `Accept` header.
+
+We can add the usage of `Accept-Encoding` HTTP header, which advertises which content type encoding the client is able to undetstand. 
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the `Accept` header is `text/plain` and `Accept-Encoding` is `base64`.
```

Comment:
@liavyona why do we need both? is the Accept header you suggest here is not the "default" one?
can we make do with only "Accept-Encoding"?

Reply:
a. Yes we can (B. Obama).
b. The two headers combined together make are more certain to confirm the user actually asked for encoded response

Reply:
I can add this to the open question and we discuss it on the presentation

Reply:
i agree with @InbalZilberman . if the user can add only one header to get the desired solution then it's better.

Reply:
i would go with `-H "Accept-Encoding: base64"`. it makes sense and is clear.